proposals/conditional-operator-access-syntax-refinement.md

```diff
@@ -0,0 +1,171 @@
+    : primary_expression '?' '.' identifier type_argument_list?
+      (null_forgiving_operator? dependent_access)*
+    ;
+null_conditional_element_access
+    : primary_expression '?' '[' argument_list ']'
+      (null_forgiving_operator? dependent_access)*
+    ;
+```
+
+But explicit rules are added stating:
+
+> `?` `.` and `?` `[` are made up of the two tokens `?` followed by either `.` or `[`. Unlike other productions in the syntactic grammar, no characters of any kind (not even whitespace) are allowed between the two tokens in each of these productions.
+
+The [Conditional Operator](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#1115-conditional-operator) section is updated to include the following rule:
+
+> A conditional expression of the form `b ? x : y` is not allowed to have `?` and `x` touch if `x` starts with a `.` or `[` token.  If `x` starts with either of those tokens, the expression will instead be interpreted as a  `null_conditional_member_access` or `null_conditional_element_access` respectively.
+
+
+### Examples
+
+**Valid null-conditional access (no space):**
+```csharp
+var x = obj?[index];        // Null-conditional indexing
+var y = obj?.Member;        // Null-conditional member access
+```
+
+**Valid conditional expressions (with space/comment):**
+```csharp
+var x = condition ? [1, 2, 3] : [4, 5, 6];    // Conditional with collection expressions
+var y = flag ? .StaticMember : expr;          // Conditional with static member access (possible future feature)
+var z = test ?/*comment*/[x] : [y];      // Comment between tokens
+```
+
+**Invalid (violates adjacency requirement):**
+```csharp
+var x = obj? [index];       // Error: space between ? and [
+var y = obj? .Member;       // Error: space between ? and .
+```
+
+## Drawbacks
+
+### Breaking change
+
+This is a breaking change. Code that currently uses whitespace between `?` and `[` for null-conditional indexing will no longer compile.  Similarly, code like `a?[b]:[c]` would break as well.  
+
+Real-world examples from existing codebases include:
+- `var aux_data = ctx? ["auxData"] as JObject;`
+- `var baseScheme = SchemeManager.GetHardCodedSchemes ()? ["Base"];`
+
+These patterns would need to be updated to remove the whitespace: `ctx?["auxData"]` and `SchemeManager.GetHardCodedSchemes()?["Base"]`.
+Similarly,  if `a?[b]:` was encountered, it would need to be updated to have explicit spaces to
+preserve parsing behavior.
+
+Note: Users using standard .Net tooling (like Visual Studio and `dotnet format`) would not be likely to run into this issue as the standard formatting engine already formats code such that it follows the syntax pattern codified here.  
+
+### Migration burden
+
+Developers will need to update their code when upgrading to the language version that includes this feature.
+
+The compiler will be augmented to parse in the following fashion:
+
+Given:
+
+```
+conditional_expression
+    : null_coalescing_expression
```

Comment:
It threw me off that `null_coalescing_expression` is used here. It might be helpful to include the following quote from the standard.

> The controlling conditional expression of the `?:` operator ([§12.19](https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/expressions.md#1219-conditional-operator)) follows the same rules as a *boolean_expression*, but for reasons of operator precedence is classified as a *null_coalescing_expression*.